app/controllers/alarms_controller.rb

```diff
@@ -1,71 +1,51 @@
 class AlarmsController < ApplicationController
   def index
-    @alarms = current_user.alarms
-    @alarms = @alarms.map do |alarm|
-      [alarm, alarm.alarmDays]
-    end
-    @alarms
+    @alarms = current_user.alarms.map {|alarm| helpers.decorate(alarm)}
   end
 
   def new
     @alarm = Alarm.new
-    @user = current_user
-    @days = create_days
   end
 
   def create
-    if params[:days].empty?
-      # message of invalid
-    else 
-      @alarm = current_user.alarms.create(alarm_params)
-      if @alarm
-        hours, minutes = params[:alarm][:time].split(':')
-        params[:days].each do |day|
-          new_time = DateTime.parse(day).change(hour: hours.to_i, min: minutes.to_i)
-          @alarm.alarmDays.create({daytime: new_time})
-          AlarmNotificationJob.set(wait_until: new_time).perform_later(current_user, @alarm)
-        end
-        redirect_to user_alarms_path(current_user)
-      else
-        render :new
+    @alarm = current_user.alarms.create(alarm_params)
+    if @alarm
```

Comment:
you could use `@alarm.vlaid?` instead of just `@alarm`, to validate it was saved correctly, maybe this is not your case actually, but using the `create` method doesn't ensure you that the record is persisted in the database.
In order to make your code less error-prone, it would be great to add this extra security step. Also if you know that the new instance is going to be manipulated in some other places before being persisted into the database you have be sure to call the `save` method on it and make this kind of error handling that you're implementing here rendering the new view again.

Reply:
You are right, it does not ensure the alarm will be created, and also I have not added error handling, thanks